tools/shareable-examples/src/utils/get-template-and-style-files-from-component-source-file.ts

```diff
@@ -0,0 +1,131 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ExampleFile } from './examples.interface';
+import {
+  SourceFile,
+  isPropertyAssignment,
+  PropertyAssignment,
+  isArrayLiteralExpression,
+} from 'typescript';
+import {
+  getAngularDecoratedClasses,
+  AngularClassDecoratorName,
+} from 'tools/shared/src/utils';
+import { promises as fs } from 'fs';
+import { resolve } from 'path';
+
+async function getDecoratorPropertyAssigments(
+  source: SourceFile,
+): Promise<PropertyAssignment[]> {
+  // Get the componentClassDecorator
+  const componentClass = (await getAngularDecoratedClasses(
+    source,
+    AngularClassDecoratorName.Component,
+  ))[0];
+  // Iterate all properties within the decorator
+  const decoratorPropertyAssignments = componentClass.decorator.properties
+    // Filter everything that is not a property assignment
+    .filter(property => isPropertyAssignment(property));
+  return decoratorPropertyAssignments as PropertyAssignment[];
+}
+
+/** Extract the templateUrl value from the decorator */
+async function getTemplatePathFromSourceFile(
+  source: SourceFile,
+): Promise<string> {
+  // Get all PropertyAssignments in the decorator
+  const decoratorPropertyAssignments = await getDecoratorPropertyAssigments(
+    source,
+  );
+
+  return (
+    decoratorPropertyAssignments
+      // Filter everything that is not the selector assignment
+      .filter(
+        (property: PropertyAssignment) =>
+          property.name.getText() === 'templateUrl',
+      )
+      // Map to the initializer value
+      .map((property: PropertyAssignment) => property.initializer.getText()) // take the first element found
+      [0] // replace the quotes
+      .replace(/'/g, '')
+  );
+}
+
+async function getStyleUrlsFromSourceFile(
+  source: SourceFile,
+): Promise<string[]> {
+  // Get all PropertyAssignments in the decorator
+  const decoratorPropertyAssignments = await getDecoratorPropertyAssigments(
+    source,
+  );
+  const stylesFilePaths = decoratorPropertyAssignments
+    // Filter everything that is not the selector assignment
+    .filter(
+      (property: PropertyAssignment) => property.name.getText() === 'styleUrls',
+    )
+    // Map to the initializer value
+    .map((property: PropertyAssignment) => {
+      if (isArrayLiteralExpression(property.initializer)) {
+        // Return every value in the array initializer
+        return property.initializer.elements.map(element => element.getText());
+      }
+    })
+    // Filter empty entries from the array.
+    .filter(Boolean)
+    // Flatten the array
+    .reduce<string[]>((aggregator, styleUrls) => {
+      return [...aggregator!, ...styleUrls!];
+    }, [])
+    // remove the quotes around the string literals
+    .map(styleUrl => styleUrl.replace(/'/g, ''));
```

Comment:
if you use the `text` property instead of the `getText` you don't have to replace anything